Upstream: https://git.reviewboard.kde.org/r/119111/
Source:   Alex Merry
Reason:   Fixes building of translations of Plasma 5 RC packages

diff --git a/cmake/modules/FindGettext.cmake b/cmake/modules/FindGettext.cmake
index 91e88f7e00ac52539066e71eeffb7df6c2148196..ff86a547c4aa875296be81d5ef08d4a26892af92 100644
--- a/cmake/modules/FindGettext.cmake
+++ b/cmake/modules/FindGettext.cmake
@@ -212,7 +212,16 @@ MACRO(GETTEXT_PROCESS_PO_FILES _lang)
       LIST(REMOVE_AT _args 0 1)
    ENDIF("${_tmp}" STREQUAL "INSTALL_DESTINATION")
    
-#    message(STATUS "2 all ${_addToAll} dest ${_installDest} args: ${_args}")
+   LIST(GET _args 0 _tmp)
+   IF("${_tmp}" STREQUAL "PO_FILES")
+      LIST(REMOVE_AT _args 0)
+   ELSE()
+      SET(_deprecation_arg DEPRECATION)
+      IF(CMAKE_VERSION VERSION_LESS 3.0.0)
+         SET(_deprecation_arg AUTHOR_WARNING)
+      ENDIF()
+      MESSAGE(${_deprecation_arg} "GETTEXT_PROCESS_PO_FILES without a PO_FILES argument is deprecated")
+   ENDIF()
    
    FOREACH(_current_PO_FILE ${_args})
       GET_FILENAME_COMPONENT(_name ${_current_PO_FILE} NAME)
